packages/features/form-builder/FormBuilder.tsx

```diff
@@ -718,6 +729,19 @@ export const FormBuilderField = ({
                 errors={formState.errors}
                 render={({ message }) => {
                   const name = message?.replace(/\{([^}]+)\}.*/, "$1");
+
+                  // For edge case when location is not hidden and required but it is not passed. (For example having only one location Attendee Phone Number in the event type)
+                  if (name !== field.name && field.name === "location" && error) {
+                    return (
+                      <div
+                        data-testid="error-message-location"
+                        className="mt-2 flex items-center text-sm text-red-700 ">
+                        <FiInfo className="h-3 w-3 ltr:mr-2 rtl:ml-2" />
+                        <p>{t("error_required_field")}</p>
+                      </div>
+                    );
+                  }
+
```

Comment:
@hariombalhara is there any better way to show this required error for location field ? location is marked as optional in zod schema.

Reply:
I don't really understand why this is required.  On production, if just Attendee Phone Number is chosen as location, it is considered required.
<img width="1187" alt="Screenshot 2023-03-15 at 11 12 27 PM" src="https://user-images.githubusercontent.com/1780212/225396806-dda3e23b-c2d9-48c0-ad20-ad76087816cf.png">

Reply:
@hariombalhara  if you completely clear attendee phone number even +91 then you won't see this error. ( this is very small edge case)

Reply:
Thanks @Udit-takkar. Didn't notice that was happening. 
Have fixed it in a better way now.